dia -v
=======
Dia version DIA_0_97_0-2406-gaa4e99118+, compiled 16:18:35 Oct 11 2021

Python 3.9.5 (default, Jun 24 2021, 13:37:02) 

source:
=======
git clone https://gitlab.gnome.org/GNOME/dia.git


Clone the git repository, for example:

 ```sh
git clone https://gitlab.gnome.org/GNOME/dia.git
#or git clone https://github.com/GNOME/dia
 ```


Open a text editor_and save this into run_with_dia_env.sh.
```sh
#!/usr/bin/env sh
if [ !  -d "$2" ] || [ ! -d "$3" ]; then
    echo Unable to find both directory $2 and $3.
fi

SOURCE_ROOT=$2
BUILD_ROOT=$3

# Windows has ; for PATH separators.
PATH_SEP=":"
if uname | grep -iq MINGW; then
    PATH_SEP=";"
fi
# This unfortunately is a duplication of run_env in
# tests/meson.build.  This is fine as long as this
# script is only used to generate run_with_dia_env
# and only that.
cat > $1 << EOF
#!/usr/bin/env sh
export DIA_BASE_PATH="${SOURCE_ROOT}"
export DIA_LIB_PATH="${BUILD_ROOT}/objects${PATH_SEP}${BUILD_ROOT}/plug-ins"
export DIA_SHAPE_PATH="${SOURCE_ROOT}/shapes"
export DIA_XSLT_PATH="${SOURCE_ROOT}/plug-ins/xslt"
export DIA_PYTHON_PATH="${SOURCE_ROOT}/plug-ins/python"
export DIA_SHEET_PATH="${BUILD_ROOT}/sheets"
\$@
EOF

chmod u+x $1

```

Run the script
```sh
sudo mkdir /opt/dia

sudo mkdir /opt/dia/plug-ins
sudo mkdir /opt/dia/shapes

sudo mkdir /opt/dia/plug-ins/xslt
sudo mkdir /opt/dia/plug-ins/python

sudo mkdir /opt/dia/sheets

./generate_run_with_dia_env.sh build/run_with_dia_env /opt/dia /opt/dia/lib
```

configure meson

 ```sh
meson setup --wipe build --prefix=/usr --buildtype=release 
#Desactive the buggy pdf, by editing plug-ins/meson.build 
#  and comment the line pdf to have #subdir('pdf')
meson setup  build --reconfigure -Dwith-python=yes
cd build &&meson --prefix=/usr --buildtype=release .. && ninja 

ninja install
```

==> Solve networ/network redundancy error: 
#in /usr/share/dia/shapes if you have both network and Network folders
#move files from network into Network and remove network, it 

